File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.util.*;
import java.io.*;
public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int left = 0;
        int answer = 0;
        String input = br.readLine();
        for(int i = 0; i< input.length(); i++){
            if(input.charAt(i) == '('){
                left++;
            }else{
                if(input.charAt(i-1) == '('){
                  left--;
                    answer += left;  
                }else{
                    
                    left--;
                    answer += 1;
                }
                
            }
        }
        System.out.println(answer);
    }
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
order = input()
total = 0
left = 0
flag = False
for ch in order:
    if ch == "(":
        left += 1
        flag = True
    else:
        if flag:
            left -= 1
            total += left
            flag = False
        else:
            left -= 1
            total += 1
        
print(str(total))